### PR TITLE
Fix Instance/Tenant Parsing for V2 Authority

### PR DESCRIPTION
**Pull request by jackj-msft**
# Fix Instance/Tenant Parsing for V2 Authority

- [x] You've read the [Contributor Guide](https://github.com/AzureAD/microsoft-identity-web/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/AzureAD/microsoft-identity-web/blob/master/CODE_OF_CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)
Fix the tenant parsing in MergedOptions for V2 authority endpoints.

## Description
Currently `ParseAuthorityIfNecessary` assumes the authority to be formatted as `{Instance}/{Tenant}`.
With the introduction of v2.0 authority endpoints, this breaks the parsing of the authority.
i.e. given the authority of `https://testinstance/tenant123/v2.0`
The existing method will return the following:
```
Instance = https://testinstance/tenant123
TenantId = v2.0
```

What we expect is the following:
```
Instance = https://testinstance
TenantId = tenant123
```

Assumptions made:
* All authority endpoints starts with `https://` (or a prefix longer than 8 characters)
* The authority is of the format `{Instance}/{Tenant}/{Version}` with the version being optional

### CODE REVIEW

File: src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs [hunk 1]
```diff
@@ -215,12 +215,12 @@ internal static void UpdateMergedOptionsFromMicrosoftIdentityOptions(MicrosoftId
             mergedOptions.TimeProvider = microsoftIdentityOptions.TimeProvider;
 #endif
 #if NET9_0_OR_GREATER
-            if (microsoftIdentityOptions.AdditionalAuthorizationParameters != null)
-            {
-                foreach (var parameter in microsoftIdentityOptions.AdditionalAuthorizationParameters)
-                {
-                    mergedOptions.AdditionalAuthorizationParameters.Add(parameter.Key, parameter.Value);
-                }
+            if (microsoftIdentityOptions.AdditionalAuthorizationParameters != null)
```
Comment (abatishchev) [inline src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs:218]: `is not null` maybe?

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @jackj-msft 

Do you have repro steps for that bug (end to end repro steps)?

**Comment by jennyf19** [comment]
@jackj-msft is there a specific version which is having the issue?

**Comment by jackj-msft** [comment]
> LGTM Thanks @jackj-msft
> 
> Do you have repro steps for that bug (end to end repro steps)?

If you specify any outbound policy that uses an AppToken AND a v2 authority endpoint (either inferred by MISE or specified)
Getting the client token will fail 
This is all outlined in the bug in ADO (bug 2986056)

> @jackj-msft is there a specific version which is having the issue?

I've been using the latest 1.27 of the Microsoft.Identity.ServiceEssentials.AspNetCore metapackage.
The dependent Microsoft.Identity.Web package is version 2.20

**Review by sruke — APPROVED** [review]

**Pull request merged**